community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/command/PhysicalLogNeoCommandReaderV2_1.java

```diff
@@ -309,7 +310,7 @@ public boolean visitRelationshipGroupCommand( Command.RelationshipGroupCommand c
             {
                 throw new IOException( "Illegal in use flag: " + inUseByte );
             }
-            int type = channel.getShort();
+            int type = shortToUnsignedInt( channel.getShort() );
```

Comment:
It is not strictly required to fix this particular reader. It is required to fix all readers for the current version. So they are `V2_2`, `V2_2_4` and `V2_2_10` for 2.2 branch. This makes it possible to start for ex. 2.2.15 on 2.2.1 non-cleanly shutdown store. I think this the main reason we need multiple log readers.